thinc/layers/parametricattention_v2.py

```diff
@@ -0,0 +1,107 @@
+from typing import Callable, Optional, Tuple, cast
+
+from ..config import registry
+from ..model import Model
+from ..types import Floats2d, Ragged
+from ..util import get_width
+
+InT = Ragged
+OutT = Ragged
+
+
+@registry.layers("ParametricAttention.v2")
+def ParametricAttention_v2(
+    *,
+    key_transform: Optional[Model[Floats2d, Floats2d]] = None,
+    nO: Optional[int] = None
+) -> Model[InT, OutT]:
+    if key_transform is None:
+        layers = []
+        refs = {}
+    else:
+        layers = [key_transform]
+        refs = {"key_transform": cast(Optional[Model], key_transform)}
+
+    layers = [key_transform] if key_transform is not None else []
```

Comment:
I was thinking, all this custom handling could be avoided if we had an identity layer. If `key_transform` is `None`, we could just use the identity layer. Then we don't need any conditionals after that (also simplifies the testing al lot).

Reply:
Good point! Isn't the `noop` basically an identity layer?

Reply:
It is, thanks :), I somehow looked over it because the standard terminology is identity.

Reply:
I'll update the PR after my current training run is done.